libraries/wutdevoptab/devoptab_fs_statvfs.c

```diff
@@ -5,7 +5,51 @@ __wut_fs_statvfs(struct _reent *r,
                  const char *path,
                  struct statvfs *buf)
 {
-   //TODO: FSGetFileSystemInfo
-   r->_errno = ENOSYS;
-   return -1;
+   FSStatus status;
+   FSCmdBlock cmd;
+   uint64_t freeSpace;
+   FSStat driveStat;
+   memset(buf, 0, sizeof(struct statvfs));
+
+   char *fixedPath = __wut_fs_fixpath(r, path);
+   if (!fixedPath) {
+      return -1;
+   }
+
+   FSInitCmdBlock(&cmd);
+
+   status = FSGetFreeSpaceSize(__wut_devoptab_fs_client, &cmd, fixedPath, &freeSpace, FS_ERROR_FLAG_ALL);
+   if (status < 0) {
+      free(fixedPath);
+      r->_errno = __wut_fs_translate_error(status);
+      return -1;
+   }
+
+   status = FSGetStat(__wut_devoptab_fs_client, &cmd, fixedPath, &driveStat, FS_ERROR_FLAG_ALL);
```

Comment:
Apparently this only works for WFS partitions? At least for /vol/external01 qutoa and alloc size is zero =(

Probably best to revert the statvfs implementation, or get the (quota) size first, and only if this succeds (returncode >=0 AND qutoasize > 0) do the (maybe) expensive FSGetFreeSpaceSize call?